### PR TITLE
Some small bugfixes and stability improvements

### PR DESCRIPTION
**Pull request by marcelveldt**
A bunch of small fixes I'd like to get in and release before merging the new features like climate and the refactored subscription logic.

- Fix devicetypes type hint
- Fix some typos 
- Add retry logic to resolving of nodes (might fix mdns resolve issues)

### CODE REVIEW

File: matter_server/server/device_controller.py [hunk 1]
```diff
@@ -632,3 +629,32 @@ def _parse_attributes_from_read_result(
...
+                )
+                return
+            LOGGER.debug("Resolving node %s", node_id)
+            await self._call_sdk(self.chip_controller.ResolveNode, nodeid=node_id)
+        except (ChipStackError, TimeoutError) as err:
+            if not retries:
+                # when we're out of retries, raise NodeNotResolving
+                raise NodeNotResolving(f"Unable to resolve Node {node_id}") from err
+            await self._resolve_node(
+                node_id=node_id, retries=retries - 1, allow_pase=retries == 0
```
Comment (MartinHjelmare) [inline matter_server/server/device_controller.py:658]:
```suggestion
                node_id=node_id, retries=retries - 1, allow_pase=retries - 1== 0
```

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

File: matter_server/server/device_controller.py [hunk 2]
```diff
@@ -421,7 +418,7 @@ async def subscribe_node(self, node_id: int) -> None:
         sub: Attribute.SubscriptionTransaction = await self.chip_controller.Read(
             nodeid=node_id,
             attributes="*",
-            events=[("*", 0)],
+            events=[("*", 1)],
```
Comment (MartinHjelmare) [inline matter_server/server/device_controller.py:421]: What does this change do?

Comment (marcelveldt) [reply matter_server/server/device_controller.py:421]: enforce urgent device events

File: pyproject.toml [hunk 3]
```diff
@@ -49,6 +49,9 @@ test = [
 [project.scripts]
 matter-server = "matter_server.server.__main__:main"
 
+[tool.codespell]
+ignore-words-list = "pase,hass"
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply pyproject.toml:53]: copy paste error, I'll adjust

File: matter_server/server/device_controller.py (hunk 1 above)
Comment (marcelveldt) [reply matter_server/server/device_controller.py:658]: good catch, we want 3 attempts total. 2 of them are without PASE and the last one is with

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**